pkg/kubectl/cmd/util/factory.go

```diff
@@ -194,6 +196,35 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 				return nil, fmt.Errorf("it is not possible to get ports from %s", mapping.Kind)
 			}
 		},
+		LabelsForResource: func(mapping *meta.RESTMapping, namespace, name string) (map[string]string, error) {
+			// TODO: replace with a swagger schema based approach (identify pod selector via schema introspection)
+			client, err := clients.ClientForVersion("")
+			if err != nil {
+				return nil, err
+			}
+			var labels map[string]string
+			switch mapping.Kind {
+			case "ReplicationController":
+				rc, err := client.ReplicationControllers(namespace).Get(name)
+				if err != nil {
+					return nil, err
+				}
+				labels = rc.Labels
+			case "Pod":
+				pod, err := client.Pods(namespace).Get(name)
+				if err != nil {
+					return nil, err
+				}
+				labels = pod.Labels
+			case "Service":
+				svc, err := client.Services(namespace).Get(name)
+				if err != nil {
+					return nil, err
+				}
+				labels = svc.Labels
+			}
```

Comment:
Add a default case. Someone might add a call from somewhere else.

It would be nice to be able to use the generic resource accessor:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/meta/meta.go#L232

But this approach, which follows the same pattern as PortsForResource, is ok for now.